src/EntityFramework/Utilities/IntegerSwitch.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Data.Entity.Utilities
+{
+    internal class IntegerSwitch : Switch
```

Comment:
@chrfin, @ajcvickers do you know how usable is System.Dianostics.Switch on .NET Core? From what I remember the last time I used one of these you configured in the app.config, but that isn't operational in .NET Core, so, are there other alternatives?

Reply:
Hmm, good point. I used a `Switch` as the class already uses `BooleanSwitch` (see `_applyTransformationsRegardlessOfSize`)...
I'll check it and if not working on .NET Core replace all (incl. the "old one") of them and update the pull-request.

Reply:
@ajcvickers @divega I checked it and Diego is right. `Switch` is useless in .NET Core as it is only working in-memory per instance.  
Now I'm not sure in which direction I should go:
1) Improve the `Switch`-implementation in CoreFX to be usable
2) Create a new `Switch`-implementation outside of CoreFX based on `Microsoft.Extensions.Configuration` (where could/should that be added? new package?)
3) Add `Microsoft.Extensions.Configuration` to EF6 and use the new config-system instead of the switches
4) Implement some global/static settings directly in EF6 changeable via code

What do you think?  
The best solution IMO would be using `Microsoft.Extensions.Configuration` in the CoreFX-`Switch`, but that is not possible AFAIK, or is it somehow?

Reply:
@ajcvickers or @bricelam may have good suggestions for this. Just some toughs: 

- We cannot add anything that depends on Microsoft.Extensions.Configuration, because the package targets .NET Standard 2.0, and doesn't support .NET Framework 4.0 and 4.5. Also, EF6's design doesn't define a good place to pass something like a configuration instance. 

- A change to System.Diagnostics.Switch that would require taking a hard dependency on System.Configuration.ConfigurationManager is unlikely to be accepted because System.Diagnostics.Switch is part of the .NET Core shared framework and this doesn't depend on ConfigurationManager by design. We also don't have a mechanism to do this with a soft dependency. 

- Elsewhere we are using AppContext.TryGetSwitch() in particular for _quirks_ you can use to opt out we usually add every time we change a behavior in a patch, but I think unfortunately that API doesn't exist prior to .NET Framework 4.6, so it wouldn't be an option for EF6. 

- All that said, EF6 already depends on System.Configuration.ConfigurationManager and we already have a few places in which we read flags from application settings:
https://github.com/aspnet/EntityFramework6/blob/8d2c24bc00aaa042a1e57eb1070f8ba23448e82a/src/EntityFramework/Core/Objects/ObjectContext.cs#L247
https://github.com/aspnet/EntityFramework6/blob/22a11d132306c257435fa7d6d75a31359cbbd05a/src/EFTools/EntityDesignerVersioningFacade/ReverseEngineerDb/SchemaDiscovery/EntityStoreSchemaGeneratorDatabaseSchemaLoader.cs#L59

I am not sure why the code in PlanCompiler uses BooleanSwitch. It is a decision that was made a long time ago when we didn't envision the possibility of running on .NET Core (or we didn't envision .NET Core for that matter :smile:), but also it sounds like a setting a user may want to change for non-diagnostic reasons.